File: reactivando-frontend/src/services/favoritoService.js
import api from './api';

export const favoritoService = {
  agregarFavorito: async (vacanteId) => {
    const response = await api.post(`/favoritos/vacante/${vacanteId}`);
    return response.data;
  },

  eliminarFavorito: async (vacanteId) => {
    await api.delete(`/favoritos/vacante/${vacanteId}`);
  },

  listarMisFavoritos: async (page = 0, size = 10) => {
    const response = await api.get(`/favoritos?page=${page}&size=${size}`);
    return response.data;
  },

  esFavorito: async (vacanteId) => {
    const response = await api.get(`/favoritos/vacante/${vacanteId}/check`);
    return response.data;
  },
};

File: reactivando-backend/src/main/java/com/reactivando/futuro/service/FavoritoService.java
package com.reactivando.futuro.service;

import com.reactivando.futuro.dto.favorito.FavoritoResponseDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface FavoritoService {
    FavoritoResponseDTO agregarFavorito(String correoCandidato, Long vacanteId);
    void eliminarFavorito(String correoCandidato, Long vacanteId);
    Page<FavoritoResponseDTO> listarMisFavoritos(String correoCandidato, Pageable pageable);
    boolean esFavorito(String correoCandidato, Long vacanteId);
}
